show/stp.py

```diff
@@ -0,0 +1,421 @@
+        
+        eth_list = [int(ifname[len("Ethernet"):]) for ifname in intf_list if ifname.startswith("Ethernet")]
+        po_list = [int(ifname[len("PortChannel"):]) for ifname in intf_list if ifname.startswith("PortChannel")]
+
+        eth_list.sort()
+        po_list.sort()
+        for port_num in eth_list:
+            ctx.invoke(show_stp_interface, ifname="Ethernet"+str(port_num))
+        for port_num in po_list:
+            ctx.invoke(show_stp_interface, ifname="PortChannel"+str(port_num))
+
+
+@show_stp_vlan.command('interface')
+@click.argument('ifname', metavar='<interface_name>', required=True)
+@click.pass_context
+def show_stp_interface(ctx, ifname):
+    """Show spanning_tree vlan interface <vlanid> <ifname> information"""
+
+    vlan_intf_tb_entry  = stp_get_entry_from_vlan_intf_tb(g_stp_appl_db, g_stp_vlanid, ifname)
+    if not vlan_intf_tb_entry:
+        return
+
+    click.echo("{:17}{:5}{:10}{:5}{:7}{:14}{:12}{:17}{:17}".format(
+        ifname,
+        vlan_intf_tb_entry['priority'],
+        vlan_intf_tb_entry['path_cost'],
+        'Y' if (stp_is_port_fast_enabled(ifname)) else 'N',
+        'Y' if (stp_is_uplink_fast_enabled(ifname)) else 'N',
+        vlan_intf_tb_entry['port_state'],
+        vlan_intf_tb_entry['desig_cost'],
+        vlan_intf_tb_entry['desig_root'],
+        vlan_intf_tb_entry['desig_bridge']
+        ))
+
+
+    pass
+
+
+@spanning_tree.command('bpdu_guard')
+@click.pass_context
+def show_stp_bpdu_guard(ctx):
+    """Show spanning_tree bpdu_guard"""
+
+    print_header = 1
+    ifname_all  = g_stp_cfg_db.get_keys("STP_PORT")
+    for ifname in ifname_all:
+        cfg_entry = g_stp_cfg_db.get_entry("STP_PORT", ifname)
+        if cfg_entry['bpdu_guard'] == 'true' and cfg_entry['enabled'] == 'true':
+            if print_header:
+                click.echo("{:17}{:13}{}".format("PortNum","Shutdown","Port Shut"))
+                click.echo("{:17}{:13}{}".format("","Configured","due to BPDU guard"))
+                click.echo("-------------------------------------------")
+                print_header=0
+
+            if cfg_entry['bpdu_guard_do_disable'] == 'true':
+                disabled = 'No'
+                keys = g_stp_appl_db.keys(g_stp_appl_db.APPL_DB, "*STP_PORT_TABLE:{}".format(ifname))
+                #only 1 key per ifname is expected in BPDU_GUARD_TABLE.
+                if keys:
+                    appdb_entry = g_stp_appl_db.get_all(g_stp_appl_db.APPL_DB, keys[0])
+                    if appdb_entry and 'bpdu_guard_shutdown' in appdb_entry:
+                        if appdb_entry['bpdu_guard_shutdown'] == 'yes':
+                            disabled = 'Yes'
+                click.echo("{:17}{:13}{}".format(ifname,"Yes",disabled))
+            else:
+                click.echo("{:17}{:13}{}".format(ifname,"No","NA"))
+
+    pass
+
+
+@spanning_tree.command('root_guard')
+@click.pass_context
+def show_stp_root_guard(ctx):
+    """Show spanning_tree root_guard"""
```

Comment:
Does this function need to show all STP enabled pots and VLAN with both "consistent" and "inconsistent" port ? just showing inconsistent ports alone is fine right? otherwise it will end up showing all VLAN, port combinations.

Also does this command take VLAN as argument? its good to have VLAN id as argument to display for specific VLAN

Reply:
This is the current design, for this release there are no plans to change this.

We will be addressing this in next release as part of IS-CLI changes
1) Display only inconsistent ports
2) Support the option of taking VLAN as argument

Reply:
ok